Tests/Integration/Network/Connection/FogSyncErrorConnectionIntTests.swift

```diff
@@ -0,0 +1,266 @@
+        expectation expect: XCTestExpectation
+    ) throws {
+        let viewIndex: UInt64 = 99
+        let ledgerIndex = viewIndex + MockFogSyncChecker.threshold + 1
+        let consensusIndex = viewIndex + MockFogSyncChecker.threshold / 2
+        
+        // at threshold, should fail
+        try attemptRefresh(
+            fogViewBlockIndex: viewIndex,
+            fogLedgerBlockIndex: ledgerIndex,
+            consensusBlockIndex: consensusIndex,
+            transportProtocol: transportProtocol,
+            expectation: expect,
+            shouldSucceed: false
+        )
+    }
+    
+    func testViewLedgerAndConsensusOverThreshold() throws {
+        try testSupportedProtocols(description: description) {
+            try viewLedgerAndConsensusOverThreshold(transportProtocol: $0, expectation: $1)
+        }
+    }
+    
+    func viewLedgerAndConsensusOverThreshold(
+        transportProtocol: TransportProtocol,
+        expectation expect: XCTestExpectation
+    ) throws {
+        let viewIndex: UInt64 = 170
+        let ledgerIndex = viewIndex - MockFogSyncChecker.threshold - 1
+        let consensusIndex = viewIndex - MockFogSyncChecker.threshold / 2
+        
+        // at threshold, should fail
+        try attemptRefresh(
+            fogViewBlockIndex: viewIndex,
+            fogLedgerBlockIndex: ledgerIndex,
+            consensusBlockIndex: consensusIndex,
+            transportProtocol: transportProtocol,
+            expectation: expect,
+            shouldSucceed: false
+        )
+    }
+    
+    func attemptRefresh(
+        fogViewBlockIndex: UInt64,
+        fogLedgerBlockIndex: UInt64,
+        consensusBlockIndex: UInt64,
+        transportProtocol: TransportProtocol,
+        expectation expect: XCTestExpectation,
+        shouldSucceed: Bool
+    ) throws {
+        let mockSyncChecker = MockFogSyncChecker(
+            viewIndex: fogViewBlockIndex,
+            ledgerIndex: fogLedgerBlockIndex,
+            consensusIndex: consensusBlockIndex)
+
+        let client = try IntegrationTestFixtures.createMobileCoinClient(
+            fogSyncChecker: mockSyncChecker,
+            transportProtocol: transportProtocol)
+
+        client.updateBalance {
+            if shouldSucceed {
+                guard $0.successOrFulfill(expectation: expect) != nil else { return }
+            } else {
+                guard $0.failureOrFulfill(expectation: expect) != nil else { return }
+            }
+
+            if let amountPicoMob = try? XCTUnwrap(client.balance.amountPicoMob()) {
+                print("balance: \(amountPicoMob)")
+                XCTAssertGreaterThan(amountPicoMob, 0)
+            }
+
+            expect.fulfill()
+        }
+    }
+    
+}
+
+class MockFogSyncChecker: FogSyncCheckable {
+    var viewsHighestKnownBlock: UInt64
+    var ledgersHighestKnownBlock: UInt64
+    var consensusHighestKnownBlock: UInt64
+
+    let fogSyncThreshold: UInt64
+    
+    static let threshold: UInt64 = 10
```

Comment:
I was thinking "user" in terms of the user of the code - even if it's us...   It's possible that logically it would never make sense to set threshold to zero - I'll review the requirements for the feature.